crates/subspace-runtime/src/lib.rs

```diff
@@ -712,7 +712,19 @@ fn extract_feeds_block_object_mapping(
     base_offset: u32,
     objects: &mut Vec<BlockObject>,
     call: &pallet_feeds::Call<Runtime>,
+    successful_calls: &mut VecDeque<Hash>,
 ) {
+    let call_hash = successful_calls.front();
```

Comment:
Smart :slightly_smiling_face: 
I'd probably go with HashSet, but this is certainly more efficient.